File: Comum/criptografia.cs
using System;
using System.IO;
using System.Security.Cryptography;

namespace Comum
{
    public class criptografia
    {
        private string sKy = "chavede0criptografia9nascomercio"; // 32 chars
        private string sIV = "vetorde9criptografia0nascomercio"; // 32 chars

        public string Criptografar(string texto)
        {
            var myRijndael = new RijndaelManaged();
            try
            {
                myRijndael.Padding = PaddingMode.Zeros;
                myRijndael.Mode = CipherMode.CBC;
                myRijndael.KeySize = 256;
                myRijndael.BlockSize = 256;

                byte[] key = System.Text.Encoding.ASCII.GetBytes(sKy);
                byte[] IV = System.Text.Encoding.ASCII.GetBytes(sIV);

                ICryptoTransform encryptor = myRijndael.CreateEncryptor(key, IV);

                var msEncrypt = new MemoryStream();
                var csEncrypt = new CryptoStream(msEncrypt, encryptor, CryptoStreamMode.Write);

                byte[] toEncrypt = System.Text.Encoding.ASCII.GetBytes(texto);
                csEncrypt.Write(toEncrypt, 0, toEncrypt.Length);
                csEncrypt.FlushFinalBlock();

                byte[] encrypted = msEncrypt.ToArray();
                return Convert.ToBase64String(encrypted);
            }
            catch
            {
                return "";
            }
            finally
            {
                myRijndael = null;
            }
        }

        public string Descriptografar(string texto)
        {
            var myRijndael = new RijndaelManaged();
            try
            {
                myRijndael.Padding = PaddingMode.Zeros;
                myRijndael.Mode = CipherMode.CBC;
                myRijndael.KeySize = 256;
                myRijndael.BlockSize = 256;

                byte[] key = System.Text.Encoding.ASCII.GetBytes(sKy);
                byte[] IV = System.Text.Encoding.ASCII.GetBytes(sIV);

                ICryptoTransform decryptor = myRijndael.CreateDecryptor(key, IV);

                byte[] sEncrypted = Convert.FromBase64String(texto);
                byte[] fromEncrypt = new byte[sEncrypted.Length + 1];

                var msDecrypt = new MemoryStream(sEncrypted);
                var csDecrypt = new CryptoStream(msDecrypt, decryptor, CryptoStreamMode.Read);

                csDecrypt.Read(fromEncrypt, 0, fromEncrypt.Length);

                return System.Text.Encoding.ASCII.GetString(fromEncrypt);
            }
            catch
            {
                return "";
            }
            finally
            {
                myRijndael = null;
            }
        }
    }
}

File: ncComum/criptografia.vb
Imports System.IO
Imports System.Security.Cryptography

Public Class criptografia

    Private sKy As String = "chavede0criptografia9nascomercio"  '32 chr shared ascii string (32 * 8 = 256 bit)
    Private sIV As String = "vetorde9criptografia0nascomercio"  '32 chr shared ascii string (32 * 8 = 256 bit)

    Public Function Criptografar(texto As String) As String
        Dim myRijndael As New RijndaelManaged

        Try
            myRijndael.Padding = PaddingMode.Zeros
            myRijndael.Mode = CipherMode.CBC
            myRijndael.KeySize = 256
            myRijndael.BlockSize = 256

            Dim encrypted() As Byte
            Dim toEncrypt() As Byte
            Dim key() As Byte
            Dim IV() As Byte

            key = System.Text.Encoding.ASCII.GetBytes(sKy)
            IV = System.Text.Encoding.ASCII.GetBytes(sIV)

            Dim encryptor As ICryptoTransform = myRijndael.CreateEncryptor(key, IV)

            Dim msEncrypt As New MemoryStream()
            Dim csEncrypt As New CryptoStream(msEncrypt, encryptor, CryptoStreamMode.Write)

            toEncrypt = System.Text.Encoding.ASCII.GetBytes(texto)

            csEncrypt.Write(toEncrypt, 0, toEncrypt.Length)
            csEncrypt.FlushFinalBlock()

            encrypted = msEncrypt.ToArray()

            Return Convert.ToBase64String(encrypted)
        Catch ex As Exception
            Return ""
        Finally
            myRijndael = Nothing
        End Try
    End Function

    Public Function Descriptografar(texto As String) As String
        Dim myRijndael As New RijndaelManaged

        Try
            myRijndael.Padding = PaddingMode.Zeros
            myRijndael.Mode = CipherMode.CBC
            myRijndael.KeySize = 256
            myRijndael.BlockSize = 256

            Dim key() As Byte
            Dim IV() As Byte

            key = System.Text.Encoding.ASCII.GetBytes(sKy)
            IV = System.Text.Encoding.ASCII.GetBytes(sIV)

            Dim decryptor As ICryptoTransform = myRijndael.CreateDecryptor(key, IV)

            Dim sEncrypted As Byte() = Convert.FromBase64String(texto)

            Dim fromEncrypt() As Byte = New Byte(sEncrypted.Length) {}

            Dim msDecrypt As New MemoryStream(sEncrypted)
            Dim csDecrypt As New CryptoStream(msDecrypt, decryptor, CryptoStreamMode.Read)

            csDecrypt.Read(fromEncrypt, 0, fromEncrypt.Length)

            Return System.Text.Encoding.ASCII.GetString(fromEncrypt)
        Catch ex As Exception
            Return ""
        Finally
            myRijndael = Nothing
        End Try
    End Function

    ''' <summary>
    ''' Recebe uma connection string com a senha criptografada (via Criptografar) e devolve
    ''' a mesma string com a senha decodificada, pronta para uso em uma conexao real.
    ''' </summary>
    Public Shared Function DecodificarConnectionString(ByVal strConn As String) As String
        Dim builder As New System.Data.Common.DbConnectionStringBuilder()
        builder.ConnectionString = strConn

        If Not builder.ContainsKey("Password") Then
            Return strConn
        End If

        Dim senhaCriptografada As String = builder("Password").ToString()

        If String.IsNullOrEmpty(senhaCriptografada) Then
            Return strConn
        End If

        Dim cripto As New criptografia()
        Dim senhaDecodificada As String = Split(cripto.Descriptografar(senhaCriptografada), vbNullChar)(0)

        Return strConn.Replace(senhaCriptografada, senhaDecodificada)
    End Function

End Class
